pkgs/by-name/py/pychess/package.nix

```diff
@@ -0,0 +1,74 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  pkg-config,
+  gobject-introspection,
+  wrapGAppsHook3,
+  gtk3,
+  gst_all_1,
+  gtksourceview,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "pychess";
+  version = "1.0.5";
+
+  src = fetchFromGitHub {
+    owner = "pychess";
+    repo = "pychess";
+    rev = "${version}";
+    sha256 = "sha256-hxc+vYvCeiM0+oOu1peI9qkZg5PeIsDMCiydJQAuzOk=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    gobject-introspection
+    wrapGAppsHook3
+  ];
+
+  buildInputs = [
+    gtk3
+    gst_all_1.gst-plugins-base
+    gtksourceview
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
+    pygobject3
+    pycairo
+    sqlalchemy
+    pexpect
+    psutil
+    websockets
+    ptyprocess
+  ];
+
+  dontWrapGApps = true;
+
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  preBuild = ''
+    export HOME=$(mktemp -d)
+    export PYTHONPATH=./lib:$PYTHONPATH
+    ${python3Packages.python.interpreter} pgn2ecodb.py
+    ${python3Packages.python.interpreter} create_theme_preview.py
+  '';
+
+  postInstall = ''
+    cp -r $out/share/pychess/* $out/lib/python*/
```

Comment:
Is this needed to be able to execute the app?

Reply:
I do believe so. Without the assets copied across, the app fails to execute and throws the following:
```
Warning: eco.db not found, please run pgn2ecodb.sh
Traceback (most recent call last):
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/bin/.pychess-wrapped", line 237, in <module>
    from pychess.Main import PyChess
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/site-packages/pychess/Main.py", line 22, in <module>
    from pychess.widgets import enginesDialog, newGameDialog
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/site-packages/pychess/widgets/enginesDialog.py", line 14, in <module>
    from pychess.widgets import newGameDialog
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/site-packages/pychess/widgets/newGameDialog.py", line 70, in <module>
    from pychess.widgets.BoardControl import BoardControl
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/site-packages/pychess/widgets/BoardControl.py", line 41, in <module>
    from . import preferencesDialog
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/site-packages/pychess/widgets/preferencesDialog.py", line 34, in <module>
    from pychess.gfx import Pieces
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/site-packages/pychess/gfx/Pieces.py", line 128, in <module>
    makruk_svg_pieces = get_svg_pieces("makruk")
                        ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/site-packages/pychess/gfx/Pieces.py", line 116, in get_svg_pieces
    rsvg_handles[c][p] = Rsvg.Handle.new_from_file(
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
gi.repository.GLib.GError: rsvg-error-quark: I/O error: Error opening file /nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/pieces/makruk/wp.svg: No such file or directory (0)
```
(In addition to the seemingly benign `ERROR: gst_player requires pygobject to be installed.`)